UI/ChatGPT/src/ChatGPT/Services/ChatService.cs

```diff
@@ -43,56 +64,61 @@ public async IAsyncEnumerable<ChatResponse> AskAsStream(ChatRequest chatRequest,
 		var response = new ChatResponse();
 		var content = new StringBuilder();
 
-		IAsyncEnumerator<ChatCompletionCreateResponse>? responseStream = default;
-		while (!response.IsError)
+		AsyncResultCollection<StreamingChatCompletionUpdate> result = _client.CompleteChatStreamingAsync(request);
+		var enumerator = result.GetAsyncEnumerator(ct);
+
+		try
 		{
-			try
+			while (await enumerator.MoveNextAsync())
 			{
-				responseStream ??= _client.CreateCompletionAsStream(request).GetAsyncEnumerator(ct);
-				if (await responseStream.MoveNextAsync())
+				var update = enumerator.Current;
+
+				try
 				{
-					if (responseStream.Current.Successful)
+					foreach (ChatMessageContentPart updatePart in update.ContentUpdate)
 					{
-						foreach (var choice in responseStream.Current.Choices)
+						try
 						{
-							content.Append(choice.Message.Content);
+							content.Append(updatePart.Text);
+							response = response with { Message = content.ToString() };
+						}
+						catch (Exception)
+						{
+							response = response with { Message = "Something went wrong.", IsError = true };
 						}
-						response = response with { Message = content.ToString() };
-					}
-					else
-					{
-						response = response with { Message = responseStream.Current.Error?.Message, IsError = true };
 					}
 				}
-				else
+				catch (Exception)
 				{
-					yield break;
+					response = response with { Message = "Something went wrong.", IsError = true };
 				}
-			}
-			catch (Exception)
-			{
-				response = response with { IsError = true };
-			}
 
-			yield return response;
+				yield return response;
+			}
+		}
+		finally
+		{
+			await enumerator.DisposeAsync();
 		}
 	}
 
-	private ChatCompletionCreateRequest ToCompletionRequest(ChatRequest request)
+	private IEnumerable<ChatMessage> ToCompletionRequest(ChatRequest request)
 	{
 		var history = request.History;
 		var requestMessages = new List<ChatMessage>(history.Count + 1)
 		{
-			ChatMessage.FromSystem(systemPrompt)
+			ChatMessage.CreateSystemMessage(systemPrompt)
 		};
-		requestMessages.AddRange(history.Select(entry => entry.IsUser
-			? ChatMessage.FromUser(entry.Message)
-			: ChatMessage.FromAssistant(entry.Message)));
 
-		return new ChatCompletionCreateRequest()
-		{
-			Messages = requestMessages,
-			Model = Models.Gpt_3_5_Turbo
-		};
+		var historyList = history.Select(ConvertMessage);
+
+		requestMessages.AddRange(historyList);
+
+		return requestMessages;
 	}
```

Comment:
```cs
private IEnumerable<ChatMessage> ToCompletionRequest(ChatRequest request)
{
	request.History
		.Select(ConvertMessage)
		.Prepend(ChatMessage.CreateSystemMessage(systemPrompt));
}
```

Reply:
you could chain .ToArray() to make it more debug-friendly.

Reply:
Thanks, it's adjusted now.